### PR TITLE
Merge Team2 tagsystem into deploy

### PR DESCRIPTION
**Pull request by symsoph**
We managed to get tags to render and save with electron. We added a message when a search by title fails: "There are no notes with this title".

### CODE REVIEW

**Review by kieraliz — APPROVED** [review]

**Comment by TerenceTan1021** [comment]
reverted the deploy branch so electron will now work for team 2 branch

**Review by dyc-github — CHANGES_REQUESTED** [review]
Please pull from deploy before your PR. Feel free to ping me on slack after and I will review it ASAP.

**Review by geenalimfat — APPROVED** [review]

File: src/scripts/render.js [hunk 1]
```diff
@@ -252,22 +664,293 @@ document.addEventListener("DOMContentLoaded", () => {
...
+    deleteActiveNote,
+    deleteNote,
+    renderNotes,
+    filterNotes,
+    createNoteElement,
+    getClosestAncestorEl,
+  };
+}
+
+module.exports = initializeNoteApp;
```
Comment (dyc-github) [inline src/scripts/render.js:956]: what is calling render.js as a module? I don't think it should have any exports as its being treated as a normal scripts file though I may be wrong.

**Review by dyc-github — DISMISSED** [review]
There's a bug where if I create a new note everything is undefinded.

File: src/scripts/render.js (hunk 1 above)
Comment (sarveshmann) [reply src/scripts/render.js:956]: That's needed for unit tests.

**Pull request merged**